### PR TITLE
bus/coco: add CoCo Max Hi Res Input Module

### PR DESCRIPTION
**Pull request by tlindner**
Add new expansion device: a mouse A/D input cartridge.
Fix SAM to not mask video RAM in ROM/RAM mode. This lets CoCo Max run correctly.

### CODE REVIEW

**Comment by tlindner** [comment]
Is that Windows test failure something I should worry about? It seems to be running out of memory during linking.

File: src/mame/machine/6883sam.cpp [hunk 1]
```diff
@@ -370,6 +381,54 @@ void sam6883_device::internal_write(offs_t offset, uint8_t data)
 		update_memory();
 	if (xorval & (SAM_STATE_R1|SAM_STATE_R0))
 		update_cpu_clock();
+
+	if( xorval & (SAM_STATE_F6|SAM_STATE_F5|SAM_STATE_F4|SAM_STATE_F3|SAM_STATE_F2|SAM_STATE_F1|SAM_STATE_F0))
```
Comment (cuavas) [inline src/mame/machine/6883sam.cpp:385]: Please don’t do weird stuff with spacing around parentheses.  Follow the format of the rest of the file.

File: src/mame/machine/6883sam.cpp [hunk 2]
```diff
@@ -370,6 +381,54 @@ void sam6883_device::internal_write(offs_t offset, uint8_t data)
...
+
+	if( xorval & (SAM_STATE_M0|SAM_STATE_M1))
+	{
+		/* Video frame buffer address changed */
+		LOGMBITS("%s: SAM M Bits: $%02x\n",
+			machine().describe_context(),
+			(m_sam_state & (SAM_STATE_M0|SAM_STATE_M1)) >> 9 );
+	}
+
+	if( xorval & (SAM_STATE_R0|SAM_STATE_R1))
```
Comment (cuavas) [inline src/mame/machine/6883sam.cpp:425]: All of these have weird inconsistent spacing.

File: src/devices/bus/coco/coco_max.cpp [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    coco_max.cpp
+
+    Code for emulating CoCo Max Hi-Res Input Module
+
+***************************************************************************/
+
+#include "emu.h"
+#include "coco_max.h"
+
+#include "machine/ram.h"
```
Comment (cuavas) [inline src/devices/bus/coco/coco_max.cpp:14]: Why do you have this here?  It appears to cause dependencies for no reason.

**Pull request merged**